### DISCUSSION TITLE
Exception when using the CLI - Context.getConnection() is null

### DISCUSSION THREAD

**Discussion by ezraaaaaaaa** [infinispan/infinispan#11670] [Q&A]
I am new to Infinispan and trying to configure it according to the documentation.
So I have set up a local instance using Docker without further configuration:

`docker run -it -p 11222:11222 -e USER="admin" -e PASS="password" quay.io/infinispan/server:14.0`

Within the container I wanted to use the CLI to configure additional users, but always get an exception when it comes to roles.
For example this command 

`/opt/infinispan/bin/cli.sh user roles ls admin`

results in

`Cannot invoke "org.infinispan.cli.connection.Connection.execute(java.util.function.BiFunction, org.infinispan.cli.connection.Connection$ResponseMode)" because the return value of "org.infinispan.cli.Context.getConnection()" is null`

Same in version 15. But creating a user works.
What am I missing here? Thanks a lot!

**Comment by ezraaaaaaaa** [comment]
Ok, a misunderstanding on my side. One needs to 'connect' first for certain commands...

**Reply by tristantarrant** [reply]
We normally protect against this using a "command activation check", but unfortunately this doesn't apply to subcommands. I'll see what I can do to improve the UX
